src/selectors/getChildren.ts

```diff
@@ -84,7 +84,7 @@ const rankDiff = (a: Child, b: Child) => Math.abs(a?.rank - b?.rank)
 const getChildrenSortedAlphabetical = (state: State, context: Context): Child[] => {
   const sorted = getChildrenSortedBy(state, context, compareThought)
   const emptyIndex = sorted.findIndex(child => !child.value)
-  return emptyIndex === -1 ? sorted : resortEmptyInPlace(sorted)
+  return emptyIndex === -1 || sorted.length === 1 ? sorted : resortEmptyInPlace(sorted)
```

Comment:
If you place the guard here, then `resortEmptyInPlace` still has a possible input that is valid to the type system but will cause the app to crash. It would be better to place the guard in `resortEmptyInPlace`.

Another consideration is module responsibility: `getChildrenSortedAlphabetical` shouldn't have to know that much about `resortEmptyInPlace`. It should only have to know that it takes `Child[]` and returns `Child[]`.

Reply:
Thanks!